lib/bowling.rb

```diff
@@ -0,0 +1,143 @@
+require 'pry'
+
+class Bowling
+  def initialize(player)
+    @player = player
+    @try_result = {try_1: 0, try_2: 0}
+    @total_score = 0
+    @frame_record = []
+    @counter = 0
+    @frame
+    @strike_counter = 0
+  end
+
+  attr_accessor :player, :try_result, :total_score, :frame, :frame_record, :counter, :strike_counter
+
+  def current_score
+    "#{player} - Frame record: #{frame_record} - Total score: #{total_score}"
+  end
+
+  def score_recorder(try_result)
+   @frame = try_result[:try_1] + try_result[:try_2]
+   if counter  == 0
+      if strike(try_result) == true
+        frame_record << 'X'
+        @counter +=1
+        @strike_counter += 1
+      elsif spare(try_result) == true
+        frame_record << 'S'
```

Comment:
The `frame_record` and `counter` code can be extracted to a private method to avoid repetition.